### ISSUE TITLE
SAML social app with `provider_id` isn't found by query in `DefaultSocialAdapter.list_apps`

### ISSUE THREAD

**Issue by agorka-contractsafe** [pennersr/django-allauth#3401]
The SAML documentation mentions the combination of the provider_id and uid needing to be unique for a social account, but if a provider_id is set in the social app, it won't be found by the query looking for DB social apps (in `DefaultSocialAdapter.list_apps`):

```
if provider:
    db_apps = db_apps.filter(
        Q(provider_id="", provider=provider) | Q(provider_id=provider)
    )
```

In that code snippet `provider` is `saml`, so if we used something like the entity ID for provider_id, the app it wouldn't be found.

For reference, here are the docs:

```
# Accounts signed up via this provider will have their
# `SocialAccount.provider` value set to this ID. The combination
# of this value and the `uid` must be unique. The IdP entity ID is a
# good choice for this.
```

I think that query may need to be modified, but also wanted to confirm whether or not the documentation still reflected how you imagined the saml provider working. Thanks!

**Comment by pennersr** [comment]
You are right, there is something wrong in that lookup. Apps can be stored either in the database, or, in the settings. And, the lookup in the settings differs from that of the DB lookup. The settings based lookup is correct -- to be fixed.

**Comment by pennersr** [comment]
Fixed via de6cea8e, released in 0.55.1

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3401
